mesonbuild/interpreter.py

```diff
@@ -2282,6 +2299,33 @@ def _get_native() -> str:
             else:
                 return _get_native()
 
+    @noArgsFlattening
+    @permittedKwargs({'native'})
+    @FeatureNew('meson.has_external_property', '0.58.0')
+    def has_external_property_method(self, args: T.Sequence[str], kwargs: dict) -> str:
+        if len(args) != 1:
+            raise InterpreterException('Must have exactly one positional argument.')
+        propname = args[0]
+        if not isinstance(propname, str):
+            raise InterpreterException('Property name must be string.')
+
+        def _get_native() -> str:
+            try:
+                props = self.interpreter.environment.properties.build
+                return propname in props
+            except Exception:
+                return False
+        if 'native' in kwargs:
+            if kwargs['native']:
+                return _get_native()
+            else:
+                return self.has_cross_property_method(args, {})
+        else:  # native: not specified
+            if self.build.environment.is_cross_build():
+                return self.has_cross_property_method(args, kwargs)
+            else:
+                return _get_native()
```

Comment:
I think you could simplify all of this to:
```python
machine = self.interpreter.machine_from_native_kwarg(kwargs)
return prop in self.interpreter.properties[machine]
```

Reply:
Hrm yes, I thought it was all a bit convoluted :laughing:

Thanks, fixed.